site/src/examples/stepper-input/ButtonAdornments.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  Button,
+  FormField,
+  FormFieldHelperText,
+  FormFieldLabel,
+  StackLayout,
+} from "@salt-ds/core";
+import { AddIcon, RefreshIcon, RemoveIcon, SyncIcon } from "@salt-ds/icons";
+import { StepperInput } from "@salt-ds/lab";
+import { useEffect, useState } from "react";
+
+const RefreshAdornment = () => {
+  const defaultValue = 1000;
+  const [value, setValue] = useState<number | string>(defaultValue);
+  return (
+    <FormField>
+      <FormFieldLabel>Refresh adornment</FormFieldLabel>
+      <StepperInput
+        value={value}
+        onChange={(_, newValue) => setValue(newValue)}
+        step={5}
+        stepBlock={50}
+        endAdornment={
+          <Button
+            onClick={() => setValue(defaultValue)}
+            appearance="transparent"
+          >
+            <RefreshIcon aria-label="Reset to default" />
+          </Button>
+        }
+      />
+      <FormFieldHelperText>Custom step 5 and step block 50</FormFieldHelperText>
+    </FormField>
+  );
+};
+
+const SyncAdornment = () => {
+  const [randomLiveValue, setRandomLiveValue] = useState(14.75);
+  const [value, setValue] = useState<number | string>(randomLiveValue);
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      const randomDelta = Math.floor(Math.random() * (1000 - 100) + 100) / 100;
+      setRandomLiveValue((prev) => prev + randomDelta);
```

Comment:
More of a note, when you sync the value you get "invalid" numbers


https://github.com/user-attachments/assets/ace30230-c566-4194-a739-4e4a50f89fa4